vendor/github.com/coredns/corefile-migration/migration/caddy/alltokens.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2015 Light Code Labs, LLC
```

Comment:
is this package a copy from a different upstream? do we have accurate licensing info for this?
is there a reason this isn't imported from the original authors? was it modified?

Reply:
^ @chrisohaver

Reply:
Originally I just imported the caddy package. But at the [suggestion](https://github.com/kubernetes/kubernetes/pull/78033#issuecomment-497464374) of reviewers concerned for the number of imported packages, I copied the required pieces locally.  The licensing was copied verbatim from the original project.  IIRC, I had to modify it slightly (a function moved from one package to another), to reduce the amount of copied code.

> do we have accurate licensing info for this (in Godeps/) ?

Sorry, I'm not familiar with the licensing process ... Please let me know what I need to do here to make everyone happy.

Reply:
> Originally I just imported the caddy package. But at the suggestion of reviewers concerned for the number of imported packages, I copied the required pieces locally. The licensing was copied verbatim from the original project. 

I see, that makes sense I suppose.

> IIRC, I had to modify it slightly (a function moved from one package to another), to reduce the amount of copied code.

So you modified one of the source files then? Do we need to denote this somehow?

> Sorry, I'm not familiar with the licensing process ... Please let me know what I need to do here to make everyone happy.

I'm not actually sure either, I haven't seen this case before.

Reply:
> So you modified one of the source files then? Do we need to denote this somehow?

I recall I had to do that to stop the "bleeding" ... otherwise i'd be duplicating most caddy.  I don't know if we need to or how to properly denote this (comments in the files? readme in same directory?).  IMHO, copying files like this feels a hack to me, and I'd prefer just importing the packages normally.  OTOH, I don't have a good understanding of the concern for importing the package in the first place, other than it created a lot of dependencies (which I infer is bad).

Reply:
cc @nikhita @justaugustus

Reply:
Is there any resolution on how we should proceed with this so that we can get this PR merged ASAP?

Reply:
I propose we just import the packages normally, and remove the copied files.  Any objections?

Reply:
@chrisohaver -- yeah, I'd feel better about that route.

Reply:
/hold
(until that's resolved.)

Reply:
I fixed the part. @justaugustus PTAL and cancel the hold if you feel it's good. Thanks!

Reply:
Thanks for fixing the licensing piece, @rajansandeep!
/hold cancel